gordo/reporters/mlflow.py

```diff
@@ -0,0 +1,440 @@
+
+            `{
+                 "subscription_id":<value>,
+                 "resource_group":<value>,
+                 "workspace_name":<value>
+             }`
+    service_principal_kwargs: dict
+        AzureML ServicePrincipalAuthentication keyword arguments. An empty dict
+        will result in interactive authentication.
+        Example::
+
+            `{
+                 "tenant_id":<value>,
+                 "service_principal_id":<value>,
+                 "service_principal_password":<value>
+             }`
+
+    Returns
+    -------
+    client: mlflow.tracking.MlflowClient
+        Client with tracking uri set to AzureML if configured.
+    """
+    logger.info("Creating MLflow tracking client.")
+
+    tracking_uri = None
+
+    # Get AzureML tracking_uri if using Azure as backend
+    if workspace_kwargs:
+        required_keys = ["subscription_id", "resource_group", "workspace_name"]
+        _validate_dict(workspace_kwargs, required_keys)
+
+        msg = "Configuring AzureML backend with {auth_type} authentication."
+        if service_principal_kwargs:
+            required_keys = [
+                "tenant_id",
+                "service_principal_id",
+                "service_principal_password",
+            ]
+            _validate_dict(service_principal_kwargs, required_keys)
+
+            logger.info(msg.format(auth_type="ServicePrincipalAuthentication"))
+            workspace_kwargs["auth"] = ServicePrincipalAuthentication(
+                **service_principal_kwargs
+            )
+        else:
+            logger.info(msg.format(auth_type="interactive"))
+            workspace_kwargs["auth"] = InteractiveLoginAuthentication(force=True)
+
+        tracking_uri = Workspace(**workspace_kwargs).get_mlflow_tracking_uri()
+
+    mlflow.set_tracking_uri(tracking_uri)
+
+    return MlflowClient(tracking_uri)
+
+
+def get_run_id(client: MlflowClient, experiment_name: str, model_key: str) -> str:
+    """
+    Get an existing or create a new run for the given model_key and experiment_name.
+
+    The model key corresponds to a unique configuration of the model. The corresponding
+    run must be manually stopped using the `mlflow.tracking.MlflowClient.set_terminated`
+    method.
+
+    Parameters
+    ----------
+    client: mlflow.tracking.MlflowClient
+        Client with tracking uri set to AzureML if configured.
+    experiment_name: str
+        Name of experiment to log to.
+    model_key: str
+        Unique ID of model configuration.
+
+    Returns
+    -------
+    run_id: str
+        Unique ID of MLflow run to log to.
+    """
+    experiment = client.get_experiment_by_name(experiment_name)
+
+    experiment_id = (
+        getattr(experiment, "experiment_id")
+        if experiment
+        else client.create_experiment(experiment_name)
+    )
+    return client.create_run(experiment_id, tags={"model_key": model_key}).info.run_id
+
+
+def _datetime_to_ms_since_epoch(dt: datetime) -> int:
```

Comment:
It seems like there should exist some function in python doing this already, no? Maybe we can handle that later

Reply:
ya maybe, but ya, it was just moved from `builder.mlflow_utils`